bin/ROC_Efficiency.cc

```diff
@@ -0,0 +1,235 @@
+////////////////////////////////////////////////////////////////////
+//
+// Dean Andrew Hidas <Dean.Andrew.Hidas@cern.ch>
+//
+// Created on: Mon Jul  4 19:20:41 CEST 2011
+//
+////////////////////////////////////////////////////////////////////
+//Edited by Thoth Gunter
+
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <cmath>
+
+#include "PLTEvent.h"
+#include "PLTU.h"
+
+
+
+class HitCounter
+{
+  public:
+    HitCounter () {
+      NFiducial[0] = 0;
+      NFiducial[1] = 0;
+      NFiducial[2] = 0;
+      NFiducialAndHit[0] = 0;
+      NFiducialAndHit[1] = 0;
+      NFiducialAndHit[2] = 0;
+    }
+    ~HitCounter () {}
+
+    int NFiducial[3];
+    int NFiducialAndHit[3];
+};
+
+
+
+// FUNCTION DEFINITIONS HERE
+int TrackingEfficiency (std::string const, std::string const, std::string const);
+
+
+
+
+
+
+
+// CODE BELOW
+
+int TrackingEfficiency (std::string const DataFileName, std::string const GainCalFileName, std::string const AlignmentFileName)
+{
+  std::cout << "DataFileName:      " << DataFileName << std::endl;
+  std::cout << "AlignmentFileName: " << AlignmentFileName << std::endl;
+
+  // Set some basic style for plots
+  PLTU::SetStyle();
+
+  // Grab the plt event reader
+  PLTEvent Event(DataFileName, GainCalFileName, AlignmentFileName);
+
+  PLTPlane::FiducialRegion FidRegionHits  = PLTPlane::kFiducialRegion_Diamond;
+  PLTPlane::FiducialRegion FidRegionTrack = PLTPlane::kFiducialRegion_m5_m5;//<-Original
+  Event.SetPlaneFiducialRegion(FidRegionHits);
+  Event.SetPlaneClustering(PLTPlane::kClustering_AllTouching,PLTPlane::kFiducialRegion_All);
+
+  PLTAlignment Alignment;
+  Alignment.ReadAlignmentFile(AlignmentFileName);
+
+  // Map for tracking efficiency
+  std::map<uint32_t, std::map<int, HitCounter> > HC;
+  
+  float const PixelDist = 5;//<-Original
+  float slope_x_low = 0.0 - 0.01;
+  float slope_y_low = 0.027 - 0.01;
+  float slope_x_high = 0.0 + 0.01;
+  float slope_y_high = 0.027 + 0.01;
```

Comment:
Although the nominal slope_y value is 0.027, if you look at the track distribution files, a lot of the scopes actually show a mean closer to 0.026, so it might be worth having a slightly wider window here to be conservative.